### PR TITLE
Fix bug that sets the range input to the resolution

### PR DESCRIPTION
**Pull request by juliusv**
This got broken as part of https://github.com/prometheus/prometheus/commit/5bcf2e6511114f8c4ee47f08296b1ff62de6a388#diff-2c68c9a4b0cb4b034e6faa8c32d3f9eeadcd60b4e5282587bf45755258ad045c.

This fixes the bug and makes the code around setting the input fields a bit more consistent.

Signed-off-by: Julius Volz <julius.volz@gmail.com>

### CODE REVIEW

**Comment by roidelapluie** [comment]
What is the user impact?

**Comment by juliusv** [comment]
@roidelapluie When you enter a number like `20` into the resolution field, then the range input field is changed to `20` as well. Although it doesn't immediately also apply that broken range from the input to what's actually used, it will confuse users...

**Review by Nexucis — COMMENTED** [review]
I don't see where is the fix.
For the moment, the only thing I see is a refactor of the code, which is ok. I just don't get how it can fix the bug you are describing @juliusv

**Comment by juliusv** [comment]
> I don't see where is the fix.

See the old line 89, new line 97. The old version changed the range input when the resolution was updated externally, the new version changes the resolution input.

File: web/ui/react-app/src/pages/graph/GraphControls.tsx [hunk 1]
```diff
@@ -58,7 +58,9 @@ class GraphControls extends Component<GraphControlsProps> {
   };
 
   changeRangeInput = (range: number): void => {
-    this.setCurrentRangeValue(formatDuration(range));
+    if (this.rangeRef.current) {
```
[the remark this answers is not in the document]
Comment (juliusv) [reply web/ui/react-app/src/pages/graph/GraphControls.tsx:61]: `null` seems to be the right value to compare to, will update.

**Comment by Nexucis** [comment]
ah yes indeed. Good catch !

All good to ship it.

**Pull request merged**